x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -235,17 +250,20 @@ private void handleLoadFailure(String modelId, Exception failure) {
 
     private void cacheEvictionListener(RemovalNotification<String, LocalModel> notification) {
         if (notification.getRemovalReason() == RemovalNotification.RemovalReason.EVICTED) {
-            String msg = new ParameterizedMessage(
+            MessageSupplier msg = () -> new ParameterizedMessage(
                 "model cache entry evicted." +
                     "current cache [{}] current max [{}] model size [{}]. " +
                     "If this is undesired, consider updating setting [{}] or [{}].",
                 new ByteSizeValue(localModelCache.weight()).getStringRep(),
                 maxCacheSize.getStringRep(),
                 new ByteSizeValue(notification.getValue().ramBytesUsed()).getStringRep(),
                 INFERENCE_MODEL_CACHE_SIZE.getKey(),
-                INFERENCE_MODEL_CACHE_TTL.getKey()).getFormattedMessage();
+                INFERENCE_MODEL_CACHE_TTL.getKey());
             auditIfNecessary(notification.getKey(), msg);
         }
+        if (notification.getRemovalReason() != RemovalNotification.RemovalReason.REPLACED) {
```

Comment:
Even if is replaced it is replaced by a model that read the old persisted stats. Should it not always persist so changes are captured.

Reply:
True.

I also don't see where in the code a cache replace is even possible. But, persisting even on replace protects us some from losing stats.